### PR TITLE
GH-226 Rebuild state when stronger chain found

### PR DESCRIPTION
**Pull request by lucafavatella**
Closes GH-226.
Supersedes PR #258.

### CODE REVIEW

File: apps/aecore/test/aec_chain_tests.erl [hunk 1]
```diff
@@ -909,31 +1012,145 @@ longest_block_chain_test_() ->
...
+               ?assertEqual({ok, {{old_top_header, HM2},
+                                  {new_top_header, HA3}}},
+                            aec_chain:force_insert_headers(AltHC)),
+
+               %% Check top changed.
+               ?assertEqual({ok, HA3}, aec_chain:top_header()),
+               ?assertEqual({ok, B1}, aec_chain:top_block()),
+
+               %% Check state of the world changed ...
+               timer:sleep(1000), %% TODO Make this event driven.
```
Comment (lucafavatella) [inline apps/aecore/test/aec_chain_tests.erl]: I understand this is because `aec_state` catches up asynchronously.

**Comment by lucafavatella** [comment]
I see CI queued. I canceled Travis CI builds related to superseded PR #258 .

**Comment by lucafavatella** [comment]
I see locally both Dialyzer and eunit tests passing.

**Comment by sennui** [comment]
both test jobs and dializer jobs green, merging

**Pull request merged**